translator.py

```diff
@@ -236,59 +249,69 @@ def _alloc_mcast_addr(self, client_addr):
-            # to) as otherwise we'll get rejected by the server.
-            form_params = {'csrfmiddlewaretoken': csrftoken, 'source': self._mcast_src_ip, 'group': str(mcast_dst_ip),
-                           'udp_port': str(self._mcast_dst_port), 'email': email, 'description': str(description),
-                           'Add': 'Add'}
-            header_fields = {'Referer': constants.MULTICASTMENU_ADD_URL}
-            resp_post = sess.post(constants.MULTICASTMENU_ADD_URL, data=form_params, headers=header_fields)
-            if resp_post.status_code != 200:
+            options = {
+                "unique_identifier": constants.MULTICASTMENU_UID,
+                "source": self._mcast_src_ip,
+                "group": str(mcast_dst_ip),
+            }
+
+            if str(src_ip) == constants.MULTICASTMENU_SENDER_IP:
+                options["inside_request"] = "1"
+
+            resp_post = requests.post(constants.MULTICASTMENU_ADD_URL, data=options)
+
+            if resp_post.status_code != 201:
                 errmsg += f'POST resulted in status code {resp_post.status_code}. Response body:\n{resp_post.text}'
                 print(errmsg)
                 return
+
+            access_code = json.loads(resp_post.text)["data"].split(" ")[-1]
+            self._acm[src_ip] = access_code
+
             msg = f'Added amt://{self._mcast_src_ip}@{mcast_dst_ip}:{self._mcast_dst_port} to the Multicast Menu ' \
-                  f'(email={email}; description={description}).'
+                  f'(source={src_ip}). The access code is {access_code}.'
             print(msg)
+
         except Exception as e:
             errmsg += f'encountered a {type(e)} with message "{e}".'
             print(errmsg)
 
-    # TODO add functionality that evicts addresses from the FIB when they've been inactive for a while.
-
+    def _identify_inactive_streams(self):
+        """
+        Loop through the streams in the FIB and determine which have not been seen in a while.
+        """
+        for src_addr in self._fib:
+            if self._loop_counter - self._atb[src_addr] > 20:
+                # Remove this stream from the FIB
+                self._allocated_mcast_addrs.remove(self._fib[src_addr])
+                # Remove the Multicast Menu entry for this stream
+                options = {
+                    "unique_identifier": constants.MULTICASTMENU_UID,
+                    "access_code": self._acm[src_addr],
+                }
+
+                resp_post = requests.post(constants.MULTICASTMENU_REMOVE_URL, data=options)
+
+                if resp_post.status_code != 201:
+                    errmsg += f'POST resulted in status code {resp_post.status_code}. Response body:\n{resp_post.text}'
+                    print(errmsg)
+                # Clean up dictionaries
+                del self._fib[src_addr]
+                del self._atb[src_addr]
+                del self._acm[src_addr]
```

Comment:
I don't think this is an issue, but let me know if my reasoning is flawed here.

When a stream is added to Multicast Menu (when it is first identified) the key is put in `_acm` in a worker thread just after the stream is added to `_fib`. This code that attempts to remove that entry from `_acm` (and `_atb` and `_fib`) won't run until at least 20 seconds after the stream is added. Additionally, the workers for adding and removing from MM are pulled from the same threadpool, meaning (I think?) that the add task is certain to be run before the remove task because it gets called first.